datasets/README.md

```diff
@@ -19,7 +19,7 @@ There are two datasets:
 - **SMCalFlow 2.0:** This is an updated version of the dataset released with the [**Task-Oriented Dialogue as Dataflow Synthesis** (TACL 2020)](https://www.mitpressjournals.org/doi/full/10.1162/tacl_a_00333) paper, which removed a very small number of incorrectly annotated examples, dropped argument names for positional arguments (so that the programs are shorter), and added inferred type arguments for type-parameterized functions that were missing in the original SMCalFlow data.
 - **TreeDST:** This is a modified version of the [TreeDST dataset]([apple/ml-tree-dst (github.com)](https://github.com/apple/ml-tree-dst)) which has been converted to the Lispress representation used for SMCalFlow 2.0, and transformed to make use of the `refer` and `revise` meta-computation operators. The transformation is described in the appendix of the paper referenced above.
 
-Furthermore, compared to the original release of the SMCalFlow dataset, these two datasets also provide programs which have been fully annotated with types for all nodes, after running a Hindley-Milner based type inference algorithm (also described in the aforementioned paper). These programs are included in the new `fully_typed_lispress` field in the JSON objects that correspond to dialogue turns. It is not recommended to use these programs directly with simple Seq2Seq baselines because they are very verbose and the information they additional information they contain can be derived directly from the `lispress` programs by running type inference. That is also why the `lispress` programs are the ones used by the official evaluation script in SMCalFlow leaderboard.
+Furthermore, compared to the original release of the SMCalFlow dataset, these two datasets also provide programs which have been fully annotated with argument names for all function arguments and types for all expressions after running a Hindley-Milner based type inference algorithm (also described in the aforementioned paper). These programs are included in the new `fully_typed_lispress` field in the JSON objects that correspond to dialogue turns. It is not recommended to use these programs directly with simple Seq2Seq baselines because they are very verbose and the information they additional information they contain can be derived directly from the `lispress` programs by running type inference. That is also why the `lispress` programs are the ones used by the official evaluation script in SMCalFlow leaderboard.
```

Comment:
Do you know if the new data works with the scripts in the codebase?
It seems that we need to add a `fully_typed_lispress` field [here](https://github.com/microsoft/task_oriented_dialogue_as_dataflow_synthesis/blob/dbad459140000c797b02c7102d791fee53a8fa5f/src/dataflow/core/dialogue.py#L70)?

I don't know whether we still support the TACL experiments with the new data. If yes, then can you or someone follow the instructions in the [README](https://github.com/microsoft/task_oriented_dialogue_as_dataflow_synthesis#smcalflow-experiments) (step 2 -- 4) and see if the instructions are still correct? Minimally, we should get the same stats as shown in the table under Step 2.

Reply:
We haven't upgraded the Program representation internally yet. We can still read the current data with the current code, it's just that fully_typed_lispress will be ignored. It would be nice if someone went through that README but I don't think we should block this PR on that of course.

Reply:
I see. When using pydantic dataclasses, loading the file would file if there are unrecognized fields. I think now we switch to the built-in dataclasses this may no longer be true. Ignoring the `fully_typed_lispress` should be fine then.